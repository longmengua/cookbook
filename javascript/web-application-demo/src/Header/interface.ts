import * as React from "react";
import {CustomRouter} from "../index_r";

export type HeaderProps = {
    children?: React.ReactNode,
    routers?: Array<CustomRouter>,
};

export type HeaderState = {

};

export const initHeaderState: HeaderState = {

};
