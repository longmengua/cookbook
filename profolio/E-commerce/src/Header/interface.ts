import * as React from "react";

export type HeaderProps = {
    children?: React.ReactNode,
};

export type HeaderState = {

};

export const initHeaderState: HeaderState = {

};
