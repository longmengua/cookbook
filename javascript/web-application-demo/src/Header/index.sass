.Header
  ul
    display: flex
    padding: 0
    margin: 0
    li
      margin: 3px 10px
      list-style: none