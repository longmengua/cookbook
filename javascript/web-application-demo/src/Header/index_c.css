.Header.css {
    text-align: left;
}