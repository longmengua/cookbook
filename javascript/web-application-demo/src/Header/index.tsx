import "./index.sass";
import "./index_c.css";
import * as React from "react";
import {useEffect, useState} from "react";
import {initHeaderState, HeaderProps, HeaderState} from "./interface";
import {Link} from "react-router-dom";

const Header: React.FC<HeaderProps> = (props: HeaderProps) => {
	const [state, setState] = useState(initHeaderState as HeaderState);
	useEffect(()=>{
		/**
		 * ... here is for componentDidMount and componentDidUpdate will trigger this
		 * */
		/**
		 * the function given to return is for componentWillUnmount()
		 * */
		return ()=> {};
	}, []);
	return (
		<div className={"Header"}>
			<ul>
				<li>
					<Link to="/">Home</Link>
				</li>
				<li>
					<Link to="/About">About</Link>
				</li>
				<li>
					<Link to="/Users">Users</Link>
				</li>
				<li>
					<Link to="/Dashboard">Dashboard</Link>
				</li>
			</ul>
		</div>
	);
};
const shouldUpdate = (prevProps: any, nextProps: any) => false;
export default React.memo(Header, shouldUpdate);
// export default React.memo(Header);
