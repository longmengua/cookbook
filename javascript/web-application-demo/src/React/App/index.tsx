import React from "react";
import "./index";
import "./index_c";

function App() {
	return (
		<div className="App">
			Index start
		</div>
	);
}

export default App;
