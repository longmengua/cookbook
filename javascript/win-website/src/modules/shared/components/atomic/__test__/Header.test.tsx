import {shallow} from "enzyme";
import Header from "../Header";
import React from "react";

describe("<Header/>", () => {
    it("take a snapshot", () => {
        const header = shallow(<Header />);
        expect(header).toMatchSnapshot();
    });
});
