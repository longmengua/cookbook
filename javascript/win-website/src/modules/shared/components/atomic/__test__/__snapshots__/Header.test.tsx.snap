// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> take a snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
